src/main/groovy/com/f2prateek/javafmt/Formatter.groovy

```diff
@@ -0,0 +1,24 @@
+package com.f2prateek.javafmt
+
+import org.gradle.api.Project
+
+class Formatter {
+  def formatterClass
+  def formatter
+
+  Formatter(Project project) {
+    def config = project.configurations.getByName(AndroidJavafmtPlugin.CONFIG_NAME)
+    def urls = new ArrayList<URL>()
+    config.files.each { File file ->
+      urls += file.toURI().toURL()
+    }
+    def cl = new URLClassLoader(urls.toArray(new URL[0]));
+
+    formatterClass = cl.loadClass("com.google.googlejavaformat.java.Formatter")
```

Comment:
I was able to reduce the amount of reflection. The plugin now has a compileOnly/provided dependency on google-java-format and directly uses it's `Formatter` class.  We still need reflection to instantiate it, afaik that is the only way to load it from the configuration that has the dependency.